### PR TITLE
Improve handling of timers and recordings with error.

### PR DESCRIPTION
**Pull request by ksooo**
@glenn-1990 could you please test. This is an alternative approach for #8

### CODE REVIEW

File: src/HTSPTypes.h [hunk 1]
```diff
@@ -151,17 +152,41 @@ struct SRecording
...
+    {
+      // determine type by state.
+      return state == PVR_TIMER_STATE_COMPLETED ||
+             state == PVR_TIMER_STATE_ABORTED   ||
+             state == PVR_TIMER_STATE_RECORDING;
+    }
+    else
+    {
+      // determine type by time (both timers and recordings may have errors).
+      return IsStopTimeInThePast();
```
Comment (Glenn-1990) [inline src/HTSPTypes.h]: Should be 'IsStartTimeInThePast()', otherwise running recordings will not show up in the "recordings" window.

Comment (ksooo) [inline src/HTSPTypes.h]: The problem with showing items with an error in the recordings listing is - at least in Confluence theme - that there is no "status" row. One cannot see that a recording has an error...

But hey, then we should make things even more simple and display everything that has state PVR_TIMER_STATE_ERROR in the timer listing - withdraw the time-based decisions in IsTimer() and IsRecording()? Everything that has an error will be a timer - as per definition. What do you think?

Comment (Glenn-1990) [inline src/HTSPTypes.h]: If using "PVR_TIMER_STATE_ERROR" only for timers, then "IsStopTimeInThePast()" is not needed anymore. I'm fine with that ;-)

Comment (ksooo) [inline src/HTSPTypes.h]: Okay. Will change that.

BTW, could you do me the favour and close #8?

**Comment by ksooo** [comment]
@Glenn-1990 Changes are in as discussed. Please test and report back.

**Comment by Glenn-1990** [comment]
keep in mind that "PVR_TIMER_STATE_ERROR" is the default 
https://github.com/kodi-pvr/pvr.hts/blob/master/src/HTSPTypes.h#L145

**Comment by Glenn-1990** [comment]
Also "PVR_TIMER_STATE_ERROR" is used for missing and invalid recording (so not only timers)

**Comment by Glenn-1990** [comment]
@ksooo 
this is working for me and based on your comments
https://github.com/kodi-pvr/pvr.hts/pull/12

**Comment by ksooo** [comment]
@Glenn-1990 Changed this pr to reflect what I think you wanted to achieve with #12. Please test and report back. #12 may be closed imo.

**Comment by ksooo** [comment]
Seeing no way to implement this in a clean way. tvheadend does not offer a suitable api at the moment. This should be addressed first. Closing this PR.

**Pull request closed without merging**